src/osyris/core/vector.py

```diff
@@ -252,3 +252,65 @@ def cross(self, other):
         z = self.x * other.y
         z -= self.y * other.x
         return self.__class__(x, y, z)
+
+    def _get_parent_pos(self):
+        group = self.parent
+        return group.get("position", group.parent["amr"]["position"])
+
+    def _get_colatitude(self, pos):
```

Comment:
`_get_colatitude` and `_get_azimuth` should probably move to the `utils`? I don't see the `self` argument used inside them.

Reply:
They're used by the `.theta` and `.phi` methods inside the if statement, where `self` is passed as a parameter.

Reply:
If `.theta` and `.phi` use `_get_colatitude` and `_get_azimuth`, we can just import those functions from `utils` (or somewhere else), my point is that `_get_colatitude` and `_get_azimuth` don't need to be part of the Vector class. They are not using any other member methods, they can be free functions.